sql/src/main/java/io/crate/operation/scalar/ArrayMergeFunction.java

```diff
@@ -0,0 +1,114 @@
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.scalar;
+
+import io.crate.metadata.*;
+import io.crate.operation.Input;
+import io.crate.planner.symbol.Function;
+import io.crate.planner.symbol.Literal;
+import io.crate.planner.symbol.Symbol;
+import io.crate.types.DataType;
+import io.crate.types.ArrayType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public abstract class ArrayMergeFunction extends Scalar<Object[], Object> {
+
+    public static final String NAME = "array_merge";
+    private FunctionInfo functionInfo;
+    private static ArrayType arrayType;
+
+    private static FunctionInfo createInfo(List<DataType> types) {
+        arrayType = (ArrayType) types.get(0);
+        return new FunctionInfo(new FunctionIdent(NAME, types), arrayType);
+    }
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(NAME, new Resolver());
+    }
+
+    protected ArrayMergeFunction(FunctionInfo functionInfo) {
+        this.functionInfo = functionInfo;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return functionInfo;
+    }
+
+    @Override
+    public Symbol normalizeSymbol(Function symbol) {
+        if (anyNonLiterals(symbol.arguments())) {
+            return symbol;
+        }
+
+        Input[] argList = new Input[symbol.arguments().size()];
+        int counter = 0;
+        for(Symbol argSymbol : symbol.arguments()){
+            Input input  = (Input) argSymbol;
+            argList[counter++] = input;
+        }
+
+        return Literal.newLiteral(evaluate(argList), arrayType);
+    }
+
+    private static class MergeFunction extends ArrayMergeFunction {
+
+        protected MergeFunction(FunctionInfo functionInfo) {
+            super(functionInfo);
+        }
+
+        @Override
+        public Object[] evaluate(Input[] args) {
+            List<Object> resultList = new ArrayList<>();
+            for(Input array : args){
+                Object[] arg = (Object[]) array.value();
+                for(Object element: arg){
+                    resultList.add(element);
+                }
+            }
+
+            return resultList.toArray();
+        }
+    }
+
+
+    private static class Resolver implements DynamicFunctionResolver {
+
+        @Override
+        public FunctionImplementation<Function> getForTypes(List<DataType> dataTypes) throws IllegalArgumentException {
+            if (dataTypes.size() < 2) {
+                throw new IllegalArgumentException("array_merge function requires at least 2 arguments");
+            } else {
+                for (int i = 0; i < dataTypes.size(); i++) {
+                    if (!(dataTypes.get(i) instanceof ArrayType)){
```

Comment:
please verify that the given array-types all have the same inner type (maybe use ``.equals``)